### DISCUSSION TITLE
[RESEARCH] Benefits of Integrating SLAM

### DISCUSSION THREAD

**Discussion by pablo-collantes** [Comet-Robotics/URC#8] [Autonomous]
**Where does this fit into the rover. Specific Team, and application on the team.** 
Helps the rover determine where it is and where it has been in its environment using an IMU (inertial measurement unit) and other sensor data.

**Describe the problem clearly** 
This approach would make our autonomous more precise. However, what are the computational drawbacks if we were to go with Multi-sensor SLAM algorithm versus other algorithms like visual or lidar? How difficult would it be to integrate to the ROS2 framework? Do the the pros out weigh the cons?

**Describe initial ideas on how to approach**
[Basic SLAM Info](https://www.mathworks.com/discovery/slam.html)
- Look at computational costs of creating 3d maps with SLAM
- Look into how we could use this with our RealSense stereoscopic camera and flag potential obstacles in the SLAM environment to go around safely.

**Comment by kevinmsmith214** [comment]
[RGB-D Slam ROS package rtabmap_slam](https://wiki.ros.org/rtabmap_slam#rtabmap)

**Reply by kevinmsmith214** [reply]
[Link showing the rtabmap_slam with a kinect on a raspberry pi](https://www.youtube.com/watch?v=c5punaP01kU)

**Comment by hectorv117** [comment]
[rtabmap_ros package doc with tutorials on how to use stero/rgbd to map and navigate outdoors ](https://wiki.ros.org/rtabmap_ros)
